pkg/framework/response.go

```diff
@@ -56,13 +56,16 @@ func (r *Response) SetSuccess(flag bool) {
 	r.success = flag
 }
 
-func (r *Response) Error(err error, code int) {
+func (r *Response) Error(err error, code int, tmplList ...string) {
 	r.err = err
+	r.statusCode = code
 
-	tmplList := []string{"./web/views/base.html",
-		"./web/views/header.html",
-		"./web/views/footer.html",
-		"./web/views/error/error.html"}
+	if len(tmplList) == 0 {
+		tmplList = []string{"./pkg/framework/view/base.html",
```

Comment:
why give the relative path to pkg? why not have the path relative to `framework`?

Reply:
if you do it relative to pkg that means you are assuming the project where this `framework` package is going to be used will have the same package structure

Reply:
not able to change the path to relative as the code automatically takes the root dir